thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/datasource/presto/SqlUtils.java

```diff
@@ -0,0 +1,386 @@
+    return "SELECT DISTINCT(" + dimension + ") FROM " + tableName + " WHERE datepartition >= daysago(1)";
+  }
+
+  private static String getSelectionClause(MetricConfigDTO metricConfig, MetricFunction metricFunction, List<String> groupByKeys, TimeGranularity granularity, TimeSpec dateTimeSpec) {
+    StringBuilder builder = new StringBuilder();
+
+    if (granularity != null) {
+      String timeFormat = dateTimeSpec.getFormat();
+      // Epoch case
+      if (timeFormat == null || TimeSpec.SINCE_EPOCH_FORMAT.equals(timeFormat)) {
+        builder.append(dateTimeSpec.getColumnName()).append(", ");
+      } else { //timeFormat case
+        builder.append(dateTimeSpec.getColumnName()).append(", ");
+      }
+  }
+
+    for (String groupByKey: groupByKeys) {
+      builder.append(groupByKey).append(", ");
+    }
+
+    String metricName = null;
+    if (metricFunction.getMetricName().equals("*")) {
+      metricName = "*";
+    } else {
+      metricName = metricConfig.getName();
+    }
+    builder.append(convertAggFunction(metricFunction.getFunctionName())).append("(").append(metricName).append(")");
+
+    return builder.toString();
+  }
+
+  static String getBetweenClause(DateTime start, DateTime endExclusive, TimeSpec timeFieldSpec, String dataset) {
+    TimeGranularity dataGranularity = timeFieldSpec.getDataGranularity();
+    long dataGranularityMillis = dataGranularity.toMillis();
+
+    String timeField = timeFieldSpec.getColumnName();
+    String timeFormat = timeFieldSpec.getFormat();
+
+    // epoch case
+    if (timeFormat == null || TimeSpec.SINCE_EPOCH_FORMAT.equals(timeFormat)) {
+      long startUnits = (long) Math.ceil(start.getMillis() / (double) dataGranularityMillis);
+      long endUnits = (long) Math.ceil(endExclusive.getMillis() / (double) dataGranularityMillis);
+
+      // point query
+      if (startUnits == endUnits) {
+        return String.format(" %s = %d", timeField, startUnits);
+      }
+
+      return String.format(" %s BETWEEN %d AND %d", timeField, startUnits, endUnits);
+    }
+
+    // NOTE:
+    // this is crazy. epoch rounds up, but timeFormat down
+    // we maintain this behavior for backward compatibility.
+    long startUnits = (long) Math.ceil(start.getMillis()) / 1000;
+    long endUnits = (long) Math.ceil(endExclusive.getMillis()) / 1000;
+
+    // point query
+    if (Objects.equals(startUnits, endUnits)) {
+      return String.format(" %s = %d", getToUnixTimeClause(timeFormat, timeField), startUnits);
+    }
+    return String.format(" %s BETWEEN %d AND %d", getToUnixTimeClause(timeFormat, timeField), startUnits, endUnits);
+  }
+
+  static String getDatePartitionClause(DateTime startTime) {
+    DateTimeFormatter inputDataDateTimeFormatter = DateTimeFormat.forPattern("yyyy-MM-dd-00");
+    return "datepartition >= " + "'" + inputDataDateTimeFormatter.print(startTime) + "'";
+  }
+
+  /**
+   * Generates PQL WHERE clause for a given filter map. The supported operation are:
```

Comment:
'SQL' instead of 'PQL'. Maybe clean up the javadocs since this is for standard sql?